_posts/2019-05-17-State TTL in Flink 1.8.0- How to Automatically Cleanup Application State in Apache Flink.md

```diff
@@ -0,0 +1,136 @@
+---
+layout: post
+title: "State TTL in Flink 1.8.0: How to Automatically Cleanup Application State in Apache Flink"
+date: 2019-05-19T12:00:00.000Z
+authors:
+- fabian:
+  name: "Fabian Hueske"
+  twitter: "fhueske"
+- andrey:
+  name: "Andrey Zagrebin"
+
+
+excerpt: A common requirement for many stateful streaming applications is to automatically cleanup application state for effective management of your state size, or to control how long the application state can be accessed. State TTL enables application state cleanup and efficient state size management in Apache Flink 
+---
+
+A common requirement for many stateful streaming applications is to automatically cleanup application state for effective management of your state size, or to control how long the application state can be accessed (e.g. due to legal regulations like the GDPR). The state time-to-live (TTL) feature was initiated in Flink 1.6.0 and enabled application state cleanup and efficient state size management in Apache Flink. 
+
+In this post, we motivate the State TTL feature and discuss its use cases. Moreover, we show how to use and configure it. We explain how Flink internally manages state with TTL and present some exciting additions to the feature in Flink 1.8.0. The blog post concludes with an outlook on future improvements and extensions.
+
+# The Transient Nature of State
+
+There are two major reasons why state should be maintained only for a limited time. For example, let’s imagine a Flink application that ingests a stream of user login events and stores for each user the time of the last login to improve the experience of frequent visitors.
+
+* Controlling the size of state
```

Comment:
We probably also need to highlight the point headers, e.g. make them bold like further for `When is the Time-to-Live reset?`:
`**Controlling the size of state.**` (and dot at the end)